File: go/raindrops/raindrops.go
package raindrops

import "fmt"

func Convert(number int) string {
	var result string
	if number%3 == 0 {
		result += "Pling"
	}
	if number%5 == 0 {
		result += "Plang"
	}
	if number%7 == 0 {
		result += "Plong"
	}
	if result == "" {
		result = fmt.Sprint(number)
	}
	return result
}

File: ruby/raindrops/raindrops.rb
# frozen_string_literal: true

# Convert an integer to Pling, Plang, Plong
class Raindrops
  def self.convert(int)
    result = ''
    result += 'Pling' if (int % 3).zero?
    result += 'Plang' if (int % 5).zero?
    result += 'Plong' if (int % 7).zero?
    result = int.to_s if result == ''
    result = 'Input must be a positive integer' if int < 1

    result
  end
end
